### ISSUE TITLE
Print correctly the caracters typed before xonsh has started

### ISSUE THREAD

**Issue by GuillaumeLeclerc** [xonsh/xonsh#959]
Hello,

Usually, since xonsh is usually slower than the other shells I tried, I end up typing before xonsh is fully started. Then my command is wrongly displayed on the terminal. The behavior is the following
- I type something, let's say 'Hello `
- Xonsh finishes loading
- The prompt line is append after `Hello`
- `Hello` is printed after the prompt line
- The cursor stays after the prompt line
- I type `how`. and since the cursor is just after the prompt line, everything under the cursor is replaced so I end up with : 

```
Hello ${PROMPT}howlo
                /\
             Cursor Here
                \/
${PROMPT}Hello how
   => This should be the correct output I think (note that the position of the cursor is correct. It's what I typed before, since it was duplicated, that shifts the text)
```

For the moment the only way I found to recover from this is to press `Ctrl-L` because everything is reprinted. Is there anything we could do on the xonsh side to fix this. `fish-shell` is behaving correctly in this situation.

**Comment by scopatz** [comment]
Crazy, I didn't know that Ctrl-L would fix this.  There are probably prompt-toolkit and readline functions that we can use to signal to do this at the end of the prompt.  A PR for this would be most welcome.

**Comment by GuillaumeLeclerc** [comment]
I think Ctrl-L would be a temporary solution.

I think we would need a more generic solution because:

- Some people would not like their screen wiped if they start xonsh (they might start it from another shell and still have somthing interesting on their screen)
- The issue occur also when you type while a command is executing. Example just do `$ sleep 5` and type `Hello` and it will have the exact same behavior.

Any idea on how we could solve that ?

**Comment by scopatz** [comment]
Potential fix in #2214

**Comment by scopatz** [comment]
Closing based on merger of #2214, please reopen if needed.
